src/components/radar-chart/radar-chart.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useCallback, useEffect, useMemo, useRef } from 'react';
+import Canvas, { CanvasRenderingContext2D } from 'react-native-canvas';
+
+import {
+  SCALE_FRACTION,
+  LINE_SIZE_FRACTION,
+  TEXT_SIZE_FRACTION,
+  TEXT_MAX_WIDTH_FRACTION,
+  TEXT_POSITION_FRACTION,
+  SCOREPOINT_SIZE_FRACTION,
+} from 'src/common/constants';
+
+import { IPoint, IPolarPoint } from 'src/common/types';
+import { getInAngleValue, polarToCartesian, normalizeAngle } from 'src/helpers';
+import { useColor } from 'src/hooks';
+
+type Axis = {
+  score: number;
+  name: string;
+};
+
+type RadarChartProps = {
+  chartSize: number;
+  maxScore: number;
+  axes: Axis[];
+};
+
+const RadarChart: React.FC<RadarChartProps> = ({
+  chartSize,
+  maxScore,
+  axes,
+}) => {
+  const canvasRef = useRef<Canvas | null>(null);
+
+  const colorHint = useColor('HINT');
+  const colorWhite = useColor('WHITE');
+  const colorBlack = useColor('BLACK');
+  const colorPrimary = useColor('PRIMARY');
+  const colorChartArea = useColor('CHART_AREA');
+
+  const halfSize = useMemo(() => chartSize / 2, [chartSize]);
+
+  const center = useMemo<IPoint>(
+    () => ({ x: halfSize, y: halfSize }),
+    [halfSize]
+  );
+
+  const chartArea = useMemo(() => chartSize * SCALE_FRACTION, [chartSize]);
+  const lineSize = useMemo(() => chartSize * LINE_SIZE_FRACTION, [chartSize]);
+
+  const textRadius = useMemo(
+    () => chartSize * TEXT_POSITION_FRACTION,
+    [chartSize]
+  );
```

Comment:
Are you sure it's good idea to use useMemo everywhere? Check this comment
<img width="921" alt="image" src="https://user-images.githubusercontent.com/70844218/160379683-77cde3a0-d417-4629-8749-8918ae6fd9c3.png">

Reply:
Yes, we can remove Memos, but Callbacks and Effects are necessary here.